specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/WebApps.json

```diff
@@ -3384,60 +3406,82 @@
         ],
         "responses": {
           "200": {
-            "description": "OK",
+            "description": "The function state was updated.",
             "schema": {
-              "$ref": "./CommonDefinitions.json#/definitions/HybridConnection"
+              "type": "string"
             }
           },
-          "default": {
-            "description": "App Service error response.",
-            "schema": {
-              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
-            }
+          "404": {
+            "description": "Function with a name of {functionName} does not exist."
           }
         }
-      },
-      "put": {
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/functions/{functionName}/properties/status": {
+      "get": {
         "tags": [
           "WebApps"
         ],
-        "summary": "Creates a new Hybrid Connection using a Service Bus relay.",
-        "description": "Creates a new Hybrid Connection using a Service Bus relay.",
-        "operationId": "WebApps_CreateOrUpdateHybridConnection",
+        "summary": "Get function status.",
+        "description": "Get function status.",
+        "operationId": "WebApps_GetFunctionStatus",
         "parameters": [
           {
             "$ref": "#/parameters/resourceGroupNameParameter"
           },
           {
             "name": "name",
             "in": "path",
-            "description": "The name of the web app.",
+            "description": "Site name.",
             "required": true,
             "type": "string"
           },
           {
-            "name": "namespaceName",
+            "name": "functionName",
             "in": "path",
-            "description": "The namespace for this hybrid connection.",
+            "description": "Function name.",
             "required": true,
             "type": "string"
           },
           {
-            "name": "relayName",
-            "in": "path",
-            "description": "The relay name for this hybrid connection.",
-            "required": true,
-            "type": "string"
+            "$ref": "#/parameters/subscriptionIdParameter"
           },
           {
-            "name": "connectionEnvelope",
-            "in": "body",
-            "description": "The details of the hybrid connection.",
-            "required": true,
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Function status returned.",
             "schema": {
-              "$ref": "./CommonDefinitions.json#/definitions/HybridConnection"
+              "$ref": "#/definitions/FunctionStatus"
```

Comment:
As it is written, both "status" and "state" are resource identifiers of the "properties" resource type. So, they should not have discrete response bodies.  It would be good to schedule a call with arm api review and discuss the properties apis, they seem problematic.

Reply:
"properties" is a virtual resource representing properties of a Function resource. To understand why these APIs are designed the way they are, some background knowledge on Azure Functions is needed. Our Function resource is not actually an ARM tracked resource. Part of our complexity in designing these APIs comes from the fact that these Function resources map to an actual live running Azure Functions host instance. To serve these ARM APIs, we have to make a proxy call to the Function App instance to query for the relevant info - much of the data cannot be stored/cached in our DB and requires these runtime proxy calls. For performance reasons, we DO cachemuch of the Function resource data - the portions that are relatively static. That way, for a list/get Functions call, we don't have to incur the overhead of a proxy call to the host runtime, and we can serve efficiently from cache. However, some information like FunctionStatus is *live* info - not a static property. To return the current status we have to call the host runtime to get the actual current status. This info cannot be cached as part of the Function resource. For list/get Functions, we also don't want to incur the overhead of proxy runtime calls to get all this info, which we'd have to do if we added these bits of info as part of the Function resource. That's why we have separate APIs for some of these.

Reply:
@mathewc can you schedule a quick call to discuss with armapireview dl?

Reply:
Removed this API from the spec update for now.